src/com/hms/flexyosisoftconnector/OsisoftJsonPayload.java

```diff
@@ -0,0 +1,397 @@
+    dataPointsEncounteredCount = 0;
+    String startOfPayload;
+
+    switch (communicationType) {
+      case OsisoftConfig.omf:
+        startOfPayload = PayloadBuilder.startOMFDataMessage();
+        break;
+      case OsisoftConfig.piwebapi:
+        startOfPayload = PayloadBuilder.startBatchOldFormat();
+        break;
+      default:
+        Logger.LOG_SERIOUS(comsErrMsg);
+        startOfPayload = "no communication type";
+        break;
+    }
+
+    boolean didAppendToPayload = payload.append(startOfPayload);
+    if (didAppendToPayload) {
+      status = PAYLOAD_HEADER_COMPLETE;
+    } else {
+      Logger.LOG_CRITICAL("An error occurred while appending the start of a payload.");
+    }
+  }
+
+  /**
+   * Append a data point to the payload
+   */
+  public void appendDataPointToPayLoad(DataPoint dataPoint) {
+
+    addDataPoint(dataPoint);
+
+    didReachMaxDataPoints();
+
+    didTakeTooLong(dataPoint);
+
+    completePayloadAttempt();
+  }
+
+  /**
+   * Ensures that we do not wait endlessly for a payload to fill up completely. Cuts off the payload
+   * after {@link PAY_LOAD_TIME_PERIOD_SECONDS}
+   *
+   * @param data
+   */
+  private void didTakeTooLong(DataPoint data) {
+    long time = Long.parseLong(data.getTimeStamp());
+
+    /*
+     * The initial value of the start time is 0.
+     * set the value to 0 whenever the payload is complete.
+     *
+     * In either case, start a new time period and set the time.
+     */
+    if (payloadStartTimestamp == 0) {
+      payloadStartTimestamp = time;
+
+    } else if (payloadStartTimestamp < (time - PAY_LOAD_TIME_PERIOD_SECONDS)) {
+      // start a new payload due to reaching time period limit
+      status = PAYLOAD_VALUES_COMPLETE;
+    }
+  }
+
+  /**
+   * Adds a datapoint to the payload
+   *
+   * @param dataPoint The data point to add
+   */
+  private void addDataPoint(DataPoint dataPoint) {
+    String newPayloadPart;
+    int index = tagIdToIndex(dataPoint.getTagId());
+
+    String timestamp = getFormattedStartTime(Long.parseLong(dataPoint.getTimeStamp()));
+    boolean isFirstPoint;
+
+    // if we have not seen that tag before, init it
+    if (checkIfTagIdEncountered(dataPoint.getTagId())) {
+      isFirstPoint = false;
+    } else {
+      // the tag has not been seen it yet
+      addToTagsEncounteredArray(dataPoint.getTagId());
+      tagPayloadArr[index] = new TagPayload();
+      ((TagPayload) tagPayloadArr[index]).dataPoint = dataPoint;
+
+      isFirstPoint = true;
+    }
+
+    switch (communicationType) {
+      case OsisoftConfig.omf:
+        newPayloadPart =
+            PayloadBuilder.addPointToOMFDataMessage(
+                dataPoint.getValueString(), timestamp, isFirstPoint);
+        break;
+      case OsisoftConfig.piwebapi:
+        newPayloadPart = PayloadBuilder.addPointOldFormat(dataPoint, dataPointsEncounteredCount);
+        break;
+      default:
+        Logger.LOG_SERIOUS(comsErrMsg);
+        newPayloadPart = "no communication type";
```

Comment:
why is this added to the payload?

Reply:
And at this point it has a communications type but its just not valid.

Reply:
I'm going to remove that line